File: SecondYear/anshika1812_AnshikaChhabra_2226CSE1186_2/Week_3/Merge Sort.c
#include <stdio.h>

int merge(int arr[],int l,int m,int r){
    int i,j,k,n1,n2,left[500],right[500];
    n1=m-l+1;
    n2=r-m;
	for (i=0;i<n1;i++)
		left[i]=arr[l+i];
	for (j=0;j<n2;j++)
		right[j]=arr[m+1+j];

	i=j=0;k=l;
	while (i<n1 && j<n2){
		if (left[i]<=right[j]){
			arr[k]=left[i];
			i++;
		}
		else {
			arr[k]=right[j];
			j++;
		}
		k++;
	}
	while(i<n1) {
		arr[k]=left[i];
		i++;
		k++;
	}
    while(j<n2) {
		arr[k]=right[j];
		j++;
		k++;
	}
    return 0;
}

int sort(int arr[],int l,int r){
    int m;
    if(l<r){
        m=(l+r)/2;
        sort(arr,l,m);
        sort(arr,m+1,r);
        merge(arr,l,m,r);
    }
    return 0;
}

int main(){
    int arr[1000],n,i;
    scanf("%d",&n);
    for(i=0;i<n;i++){
        scanf("%d",&arr[i]);
    }
    sort(arr, 0, n-1);
    for(i=0;i<n;i++){
        printf("%d ",arr[i]);
    }
    return 0;
}

File: SecondYear/Parth27904_ParthAgarwal_2226CSE(AI&ML)1015_2/Week_3/mergesort.java
import java.util.ArrayList;

public class mergesort {

    public static void ms(int[] arr,int low,int high){

        if(low>=high) return;

        int mid=(low+high)/2;

        ms(arr,low,mid);

        ms(arr,mid+1,high);

        merge(arr,low,mid,high);

    }

    public static void merge(int[] arr,int low,int mid,int high){

        int left=low;

        int right=mid+1;

 

        ArrayList<Integer> temp=new ArrayList<>();

 

        while(left<=mid && right<=high){

            if(arr[left]<arr[right]){

                temp.add(arr[left]);

                left++;

            }

            else{

                temp.add(arr[right]);

                right++;

            }

        }

        while(left<=mid){

            temp.add(arr[left]);

            left++;

        }

        while(right<=high){

            temp.add(arr[right]);

            right++;

        }

 

        for(int i=low;i<=high;i++){

            arr[i]=temp.get(i-low);

        }

 

    }

    public static void mergeSort(int[] arr, int l, int r){

        // Write your code here

        ms(arr,l,r);

    }

}